src/SignalR/common/testassets/Tests.Utils/VerifyNoErrorsScope.cs

```diff
@@ -34,6 +34,33 @@ public void Dispose()
 
             var results = _sink.GetLogs().Where(w => w.Write.LogLevel >= LogLevel.Error).ToList();
 
+            #region RemoveIn3_0
+            // -- Remove this code after 2.2 --
+            // This section of code is resolving test flakiness caused by a race in LongPolling
+            // The race has been resolved in version 3.0
+            // The below code tries to find is a DELETE request has arrived from the client before removing error logs associated with the race
+            // We do this because we don't want to hide any actual issues, but we feel confident that looking for DELETE first wont hide any real problems
+            var foundDelete = false;
+            var allLogs = _sink.GetLogs();
+            for (var i = 0; i < allLogs.Count; i++)
```

Comment:
Why not foreach?

Reply:
I probably originally wrote it differently and needed the index, but looks like I can change it to foreach now :+1: